modules/shortcodes/instagram.php

```diff
@@ -147,13 +220,22 @@ function jetpack_instagram_oembed_fetch_url( $provider, $url ) {
 		return $provider;
 	}
 
-	// Attempt to clean query params from the URL since Facebook's new API for Instagram
-	// embeds does not like query parameters. See p7H4VZ-2DU-p2.
-	$parsed_url = wp_parse_url( $url );
-	if ( $parsed_url ) {
-		$clean_url            = 'https://www.instagram.com' . $parsed_url['path'];
+	// Get a set of URL and parameters supported by Facebook.
+	$clean_parameters = jetpack_instagram_get_whitelisted_parameters( $url );
```

Comment:
```suggestion
	$clean_parameters = jetpack_instagram_get_allowed_parameters( $url );
```

(see https://github.com/Automattic/jetpack/pull/17700/files#r516918779)